pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -112,6 +121,41 @@ func loadAssignment(c *EdsCluster) *xdsapi.ClusterLoadAssignment {
 	return c.LoadAssignment
 }
 
+func serviceEntry2Endpoint(UID, address string, port uint32) *endpoint.LbEndpoint {
+	addr := core.Address{
+		Address: &core.Address_SocketAddress{
+			SocketAddress: &core.SocketAddress{
+				Address: address,
+				PortSpecifier: &core.SocketAddress_PortValue{
+					PortValue: port,
+				},
+			},
+		},
+	}
+	ep := &endpoint.LbEndpoint{
+		Endpoint: &endpoint.Endpoint{
+			Address: &addr,
```

Comment:
There is a UDS test which was still using v1 - I fixed it for v2 and is passing, i.e. the 'pipe' field is set.  It works because UDS is implemented via ServiceEntry - which is treated as config ( results in full push ). When we optimize ServiceEntry  - the test will hopefully avoid this mistake, but ServiceEntries have config info that require full push so it's a bit trickier to optimize, so left for master.

Reply:
thats not what I meant.. the addr here is obtained directly from the service entry.. and the addr in service entry is a string (`unix://...` or IP). So, if there is a service entry with unix://, you will end up sticking that string as is into a field that envoy expects to be IP.

Reply:
Got it - added family to the endpoint.